### ISSUE TITLE
Masking fails to block zeros padding

### ISSUE THREAD

**Issue by sun-peach** [keras-team/keras#13211]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu 16.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.13
- Keras version:  2.2.4
- Python version:  3.5
- CUDA/cuDNN version:  
- GPU model and memory:  2080 ti

***
Hi, dear all,

I am doing sequence to sequence modeling. Within my model, there are two sub-models, each one is dealing with recurrent network. Since my input has zeros padding, I need do masking. My model architecture is like:

input -> sub-model-1 -> sub-model-2 -> output

In sub-model-1, it is like:

input = Input((row1,col1))
temp = Masking()(input)
temp = Bidirectional(LSTM(128))(temp)
output = Bidirectional(LSTM(128))(temp)
model = Model(input,output)

In sub-model-2, it is like:
input = input(row2,col2))
y = input(rowy,coly))
temp = MHA(input, y)
temp = timedistributed(Dense(128))(temp)
output = timedistributed(softmax(31))(temp)
model = Model(inputs=[input, y], outputs=output)

The MHA here is multi-head-attention layer, which has masking internally to do attention only on the non-zero padding parts. However, I found that, the output of the sub-model-1 has non-zero entities in the zeros padding parts, which will lead the masking fail in the multi-head-attention layer. 

Why do I have the non-zero elements in the zero-padding parts from sub-model-1? Isn't the Masking layer supposed to block the padding elements? How should I fix this problem?

Thank you very much!
